File: src/0601-0700/605 - Can Place Flowers/can_place_flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: list[int], n: int) -> bool:
        empty = 0 if flowerbed[0] else 1
        for f in flowerbed:
            if f:
                n -= int((empty - 1) / 2)
                empty = 0
            else:
                empty += 1

        n -= empty // 2
        return n <= 0

File: src/0601-0700/605 - Can Place Flowers/can_place_flowers.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, mut n: i32) -> bool {
        let mut empty = if flowerbed[0] == 0 { 1 } else { 0 };

        for &x in &flowerbed {
            if x == 0 {
                empty += 1;
            } else {
                n -= (empty - 1) / 2;
                empty = 0;
            }
        }

        n -= empty / 2;

        n <= 0
    }
}
